### PR TITLE
[6.5.x] RHBPMS-4275: Guided Decision Table: XLS can not be validated

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/RHBPMS-4275

### CODE REVIEW

**Comment by manstis** [comment]
Part of an ensemble:

- https://github.com/droolsjbpm/guvnor/pull/365
- https://github.com/droolsjbpm/kie-wb-common/pull/476
- https://github.com/droolsjbpm/drools-wb/pull/298

**Comment by manstis** [comment]
@romartin Failure is due to https://github.com/uberfire/uberfire/pull/509#issuecomment-248337416

Looking into including a fix with this PR.

File: kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/validation/asset/ValidatorBuildService.java [hunk 1]
```diff
@@ -16,70 +16,131 @@
...
-                                                        final String content ) throws NoProjectException {
-        ValidatorResultBuilder resultBuilder = new ValidatorResultBuilder();
+    private List<ValidationMessage> doValidation( final Path resourcePath,
+                                                  final InputStream inputStream ) throws NoProjectException {
+        final ValidatorResultBuilder resultBuilder = new ValidatorResultBuilder();
+        final Builder builder = builderCache.assertBuilder( project( resourcePath ) );
+
+        if ( !builder.isBuilt() ) {
+            builder.build();
+        }
```
Comment (karreiro) [inline kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/validation/asset/ValidatorBuildService.java]: I think that we're building twice when the incremental build is not possible and the builder is not built. Should we avoid that?

File: kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/validation/asset/ValidatorBuildService.java [hunk 2]
```diff
@@ -16,70 +16,131 @@
...
-            resultBuilder.add( buildService.build( project( resourcePath ), resourcePath, content ).getMessages() );
+            resultBuilder.add( clone.build().getMessages() );
         }
 
         return resultBuilder.results();
     }
 
-    private boolean incrementalBuild( final Path resourcePath ) throws NoProjectException {
-        final boolean alreadyBuilt = buildService.isBuilt( project( resourcePath ) );
+    private boolean isIncrementalBuildPossible( final Path resourcePath ) throws NoProjectException {
```
Comment (karreiro) [inline kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/validation/asset/ValidatorBuildService.java]: :clap:

File: kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/validation/asset/ValidatorBuildService.java (hunk 1 above)
Comment (manstis) [inline kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/validation/asset/ValidatorBuildService.java]: @karreiro Fixed. Please re-review. Thanks, Mike

**Comment by karreiro** [comment]
+1

**Comment by romartin** [comment]
Thanks @manstis !

**Pull request merged**